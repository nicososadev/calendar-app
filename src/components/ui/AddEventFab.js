import React from 'react'
import { useDispatch } from 'react-redux'
import { uiOpenModal } from '../../actions/ui'

export const AddEventFab = () => {

    const dispatch = useDispatch()

    const handleNewEvent = () => {

        dispatch( uiOpenModal() )
    }

    return (

        <div>
            <button
                className="btn btn-primary fab"
                onClick={handleNewEvent}
            
            >
            <i className="fas fa-plus"></i>

            </button>
        </div>
    )
}
