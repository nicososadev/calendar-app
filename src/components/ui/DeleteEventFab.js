import React from 'react'
import { useDispatch } from 'react-redux'
import { deleteEventManager } from '../../actions/events'

export const DeleteEventFab = () => {

    const dispatch = useDispatch()

    const handleDeleteEvent = () => {

        dispatch( deleteEventManager() )
    }

    return (

        <div>
            <button
                className="btn btn-danger fab-danger"
                onClick={handleDeleteEvent}
            
            >
            <i className="fas fa-trash"></i>

            </button>
        </div>
    )
}
