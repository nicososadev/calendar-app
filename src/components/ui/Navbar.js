import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { logoutManager } from '../../actions/auth'

export const Navbar = () => {

    const dispatch = useDispatch()

    const handleLogout = () => {

        dispatch( logoutManager() )
    }

    const { name } = useSelector(state => state.auth)

    return (
        <div className="navbar navbar-dark bg-dark mb-4">
            <span className="navbar-brand">{ name }</span>

            <button onClick={handleLogout} className="btn btn-danger">
                <i className="fas fa-sign-out-alt" ></i>
                <span> Salir</span>
            </button>
        </div>

    )
}
