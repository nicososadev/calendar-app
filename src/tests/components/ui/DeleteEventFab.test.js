import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = {}

let store = mockStore(initialState)

store.dispatch = jest.fn()

const wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab />
    </Provider>
)

describe('Pruebas en componente DeleteEventFab', () => {
    
    test('Debe mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot()
    })
    
})
