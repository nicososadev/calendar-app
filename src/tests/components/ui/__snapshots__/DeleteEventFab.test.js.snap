// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente DeleteEventFab Debe mostrarse correctamente 1`] = `
<div>
  <button
    className="btn btn-danger fab-danger"
    onClick={[Function]}
  >
    <i
      className="fas fa-trash"
    />
  </button>
</div>
`;
